src/core/connection/SSDConverter.hpp

```diff
@@ -0,0 +1,211 @@
+#pragma once
+
+#include "base/Qv2rayBase.hpp"
+#include "common/QvHelpers.hpp"
+
+namespace Qv2ray::core::handlers
+{
+    /**
+     * A Naive SSD Decoder for Qv2ray
+     *
+     * @author DuckSoft <realducksoft@gmail.com>
+     * @copyright Licensed under GPLv3.
+     */
+    namespace ssd
+    {
+        /**
+         * pattern for the nodes in ssd links.
+         *
+         * %1: airport name
+         * %2: node name
+         * %3: rate
+         */
+        inline auto DEFAULT_NAME_PATTERN = "%1 - %2 (rate %3)";
+
+        /**
+         * @brief decodeSSD
+         * @param uri     the uri of ssd link.
+         * @param pattern the pattern for node names.
+         * @return tuple of:
```

Comment:
A tuple may have more than 2 values, why using a nested tuple here?

Reply:
when there's no need to go further, you know what to do.